modules/engine/src/model/model.ts

```diff
@@ -601,6 +616,10 @@ export class Model {
         this.pipeline.shaderLayout,
         this.bufferLayout
       );
+
+      for (const shader of prevShaders) {
+        this.shaderFactory.release(shader);
+      }
```

Comment:
Wouldn't just having two temp variables would be more readable than using an array? Sometimes simple is best...
```suggestion
      for (const shader of prevShaders) {
        this.shaderFactory.release(previousVs);
        this.shaderFactory.release(previousFs);
      }
```

Reply:
Done!